cmd/dockerd/daemon.go

```diff
@@ -227,13 +228,18 @@ func (cli *DaemonCli) start(opts daemonOptions) (err error) {
 
 	name, _ := os.Hostname()
 
+	// Use a buffered channel to pass changes from store watch API to daemon
+	// A buffer allows store watch API and daemon processing to not wait for each other
+	watchStream := make(chan *swarmapi.WatchMessage, 32)
```

Comment:
I meant interface in docker's `cluster` pkg like `cluster.Backend` `cluster.NetworkSubnetsProvider` that would be implemented by `daemon.Daemon`